File: src/main/kotlin/restart/language/psi/RestartDeclareStatement.java
package restart.language.psi;

import com.intellij.psi.PsiElement;
import org.jetbrains.annotations.NotNull;
import restart.language.psi_node.RestartDeclareItemNode;
import restart.language.psi_node.RestartIdentifierNode;

public interface RestartDeclareStatement extends PsiElement {
    @NotNull
    RestartIdentifierNode getDeclareKeyword();

    @NotNull
    RestartDeclareKey getDeclareKey();

    @NotNull
    RestartIdentifierNode[] getModifiers();

    @NotNull
    RestartDeclareBlock getDeclareBlock();

    @NotNull
    RestartDeclareItemNode[] getDeclareItemList();

    @NotNull
    getEnumerationVariant()
}

File: src/main/kotlin/restart/language/psi_node/RestartDeclareStatement.kt
@file:JvmName("RestartDeclareStatement")

package restart.language.psi_node

import com.intellij.psi.PsiElement
import restart.language.symbol.RestartSymbolKind

interface RestartDeclareStatement : PsiElement {
    val declareKeyword: RestartIdentifierNode
    val declareKey: RestartDeclareKeyNode
    val kind: RestartSymbolKind
    val modifiers: Array<RestartIdentifierNode>
    val declareBlock: RestartDeclareBlockNode
    val declareItemList: Array<RestartDeclareItemNode>
    val enumerationVariant: Array<RestartIdentifierNode>
    val aliases: Array<RestartIdentifierNode>
}